course_content/solutions/numpy_exercise_9.ipynb

```diff
@@ -0,0 +1,183 @@
+    "  * What special masked array creation routines exist to do this kind of thing more efficiently?\n",
+    "    * HINT: look up `np.ma.masked_where` and related routines.\n",
+    "  * Use `np.ma.filled()` to create a 'plain' (i.e. unmasked) array from a masked array.\n",
+    "  * How can you create a plain array from a masked array, but using a _different_ fill-value for masked points?\n",
+    "  * Try performing a mathematical operation between two masked arrays. What happens to the 'fill_value' properties when you do so?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "eleven = np.arange(0,12)\n",
+    "print(eleven)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "mask_eleven = ma.masked_array(eleven, mask=eleven<5)\n",
+    "mask_eleven"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "mask_negative = ma.masked_array(eleven, mask=eleven<1)\n",
+    "mask_negative.set_fill_value(-1.0)\n",
+    "mask_negative"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "eleven_where = ma.masked_where(eleven<1, eleven)\n",
+    "eleven_where"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "mask_negative.filled()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "eleven_where.filled()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "eleven_where.filled(598)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "sum_mask = eleven_where + mask_negative\n",
+    "sum_mask"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "mask_80 = ma.masked_array(eleven, mask=eleven<1)\n",
+    "mask_80.set_fill_value(80)\n",
+    "mask_80"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "sum_80 = mask_80 + mask_negative\n",
+    "sum_80"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "sum_80_rev = mask_negative + mask_80\n",
+    "sum_80_rev"
+   ]
+  }
```

Comment:
Maybe add a raw cell stating `fill_value takes on the value of the first masked_array in the mathematical operation.`